Workshop Script Update

frank-aleo opened the pull request:
### Why

Added more expressive scripts that exposes echo commands being run and brief explanations for inputs and outputs. It doesn't take the place of the README.md (and eventual tutorial in the documentation), but offers much more of a standalone narrative in the terminal. The user can run the entire script, and then read their terminal to see commands and outputs without hopping around too much.

### Changes

- Added more expressive, narrative scripts, with the exception of battleship, since it's much more complex and the README.md does a better job.
- Also added fixes on self.caller -> self.signer in the battleship example to ensure records are owned by the signer not the program owner.
- Scripts have all been modified to use inputs inline, so when we remove input files, the scripts are independent from them and will not break.
- All README.mds have been updated to the most current tutorial documentation (https://github.com/AleoHQ/welcome/pull/295)

### Testing

All scripts have been tested locally. They compile and run end to end without errors.

🎉

collinc97 commented on auction/README.md:
```diff
@@ -20,45 +20,146 @@ We make following assumptions about the auction:
+## Setup
+
+### 1. Install Leo
+Follow the [Leo Installation Instructions](./01_installation.md).
```
This file does not exist.
Where should this link to?

collinc97 commented on auction/README.md:
```diff
@@ -20,45 +20,146 @@ We make following assumptions about the auction:
+Follow the [Leo Installation Instructions](./01_installation.md).
+
+### 2. Download the auction example code
+Clone the source code for the auction example from [Github](https://github.com/AleoHQ/workshop/tree/master/auction).
```
What is the purpose of linking this example again?
If a developer is following this tutorial, then they have already found this link.

collinc97 commented on auction/README.md:
```diff
@@ -20,45 +20,146 @@ We make following assumptions about the auction:
+
+## How to Run
+
+The example should work out of the box. We recommend running the entire script, and then creating your own accounts to replicate the functionality. Run the entire script:
```
```suggestion
This auction program can be run using the following bash script.
Locally, it will execute Leo program functions to conduct, bid, and close a three party auction.
```
We should explain what the run script does

Simplify

collinc97 commented on auction/run.sh:
```diff
@@ -6,26 +6,28 @@ then
 ###############################################################################
 ########                                                               ########
-########            STEP 0: Initialize a new 2-party auction           ########
+########               Initialize a new 2-party auction                ########
```
Why remove the steps here?
I think it is additive to the dev experience to follow along with the steps in the script and then try with theirs locally.

collinc97 reviewed (changes requested):
Auction example feedback

collinc97 commented:
@frank-aleo Can you remove the input files for the workshop examples in this PR as well?
This will prevent downstream breakages over the next Leo release

frank-aleo commented:
Appreciate the comments and review @collinc97.

1. Added steps back to the `README.md` so it matches run scripts.
2. Simplified "how to run", added correct links, added explainers.
3. Removed all input files from examples.
